File: tradeit-android-sdk/src/main/java/it/trade/android/sdk/internal/LinkedBrokersParcelableList.java
package it.trade.android.sdk.internal;

import java.util.ArrayList;
import java.util.List;

import it.trade.android.sdk.model.TradeItLinkedBrokerParcelable;


public class LinkedBrokersParcelableList extends ArrayList<TradeItLinkedBrokerParcelable> {

    public LinkedBrokersParcelableList(List<TradeItLinkedBrokerParcelable> linkedBrokers) {
        this.addAll(linkedBrokers);
    }

    public boolean containsSameAccounts(TradeItLinkedBrokerParcelable linkedBrokerParcelable) {
        for(TradeItLinkedBrokerParcelable o : this) {
            if(o != null && o.equals(linkedBrokerParcelable) && o.equalsAccounts(linkedBrokerParcelable)) {
                return true;
            }
        }
        return false;
    }
}

File: tradeit-android-sdk/src/main/kotlin/it/trade/android/sdk/internal/LinkedBrokersParcelableList.kt
package it.trade.android.sdk.internal

import it.trade.android.sdk.model.TradeItLinkedBrokerParcelable
import java.util.*


class LinkedBrokersParcelableList(linkedBrokers: List<TradeItLinkedBrokerParcelable>) : ArrayList<TradeItLinkedBrokerParcelable>() {

    init {
        this.addAll(linkedBrokers)
    }

    fun containsSameAccounts(linkedBrokerParcelable: TradeItLinkedBrokerParcelable): Boolean {
        return this.any {
            it == linkedBrokerParcelable && it.equalsAccounts(linkedBrokerParcelable)
        }
    }
}
